### PR TITLE
feat: External Input support in SDK

### PR DESCRIPTION
**Pull request by tomerlichtash**
## Related Issues

Fixes <link_to_github_issue>

## Related PRs

| branch       | PR         |
| ------------ | ---------- |
| service a PR | Link to PR |
| service b PR | Link to PR |

## Description

A few sentences describing the overall goals of the pull request's commits.

## Must

- [ ] Tests
- [ ] Documentation (if applicable)

### CODE REVIEW

File: packages/sdks/web-component/src/lib/descope-wc/DescopeWc.ts [hunk 1]
```diff
@@ -842,19 +843,25 @@ class DescopeWc extends BaseDescopeWc {
...
+        if (this.validateOnBlur) {
+          handleReportValidityOnBlur(this.rootElement);
+        }
+      });
 
-      if (this.validateOnBlur) {
-        handleReportValidityOnBlur(this.rootElement);
-      }
+      // we need to wait for all components to render before we can set its value
+      setTimeout(() => {
```
Comment (nirgur) [inline packages/sdks/web-component/src/lib/descope-wc/DescopeWc.ts:862]: Why do we need 2 `setTimeout`? can we have just one?

File: packages/sdks/web-component/src/lib/descope-wc/DescopeWc.ts [hunk 2]
```diff
@@ -974,6 +981,39 @@ class DescopeWc extends BaseDescopeWc {
     }
   }
 
+  #updateExternalInputs() {
+    clearPreviousExternalInputs();
```
Comment (nirgur) [inline packages/sdks/web-component/src/lib/descope-wc/DescopeWc.ts:985]: Let's add a comment why this is needed

File: packages/sdks/web-component/src/lib/descope-wc/DescopeWc.ts [hunk 3]
```diff
@@ -974,6 +981,39 @@ class DescopeWc extends BaseDescopeWc {
...
+    const eles = this.rootElement.querySelectorAll('[external-input="true"]');
+    eles.forEach((ele) => this.#handleExternalInputs(ele));
+  }
+
+  #handleExternalInputs(ele: Element) {
+    if (!ele) {
+      return;
+    }
+
+    if (ele.getAttribute('external-input') !== 'true') {
```
Comment (nirgur) [inline packages/sdks/web-component/src/lib/descope-wc/DescopeWc.ts:996]: Is this needed? the selector is for `[external-input="true"]`

File: packages/sdks/web-component/test/descope-wc.test.ts [hunk 4]
```diff
@@ -334,7 +334,10 @@ describe('web-component', () => {
     await waitFor(() => screen.getByShadowText('It works!'), {
       timeout: WAIT_TIMEOUT,
     });
-    expect(autoFocusSpy).toBeCalledWith(expect.any(HTMLElement), true, true);
+
+    await waitFor(() =>
```
Comment (nirgur) [inline packages/sdks/web-component/test/descope-wc.test.ts:338]: Why this test was changed? 
Can we keep the current behavior?

File: packages/sdks/web-component/test/descope-wc.test.ts [hunk 5]
```diff
@@ -3988,11 +3997,13 @@ describe('web-component', () => {
 
     fireEvent.blur(emailInput);
 
-    await waitFor(() =>
-      expect(
-        (<HTMLInputElement>emailInput).reportValidity,
-      ).toHaveBeenCalledTimes(1),
-    );
+    await waitFor(() => {
+      setTimeout(() => {
```
Comment (nirgur) [inline packages/sdks/web-component/test/descope-wc.test.ts:4001]: Why do we need setTimeout inside waitFor?

File: packages/sdks/web-component/src/lib/descope-wc/DescopeWc.ts (hunk 2 above)
Comment (tomerlichtash) [reply packages/sdks/web-component/src/lib/descope-wc/DescopeWc.ts:985]: Added.

File: packages/sdks/web-component/src/lib/descope-wc/DescopeWc.ts (hunk 3 above)
Comment (tomerlichtash) [reply packages/sdks/web-component/src/lib/descope-wc/DescopeWc.ts:996]: Removed.

File: packages/sdks/web-component/test/descope-wc.test.ts (hunk 4 above)
Comment (tomerlichtash) [reply packages/sdks/web-component/test/descope-wc.test.ts:338]: auto-focus is not async, as it needs to wait for external inputs, so we have to `waitFor` it in the test.

File: packages/sdks/web-component/test/descope-wc.test.ts (hunk 5 above)
Comment (tomerlichtash) [reply packages/sdks/web-component/test/descope-wc.test.ts:4001]: Removed timeout and moved the blur trigger inside the `waitFor` block.

**Review by nirgur — APPROVED** [review]

**Pull request merged**